### PR TITLE
CI: Generate release tarball with vendored dependencies

### PR DESCRIPTION
**Pull request by CyberTailor**
For better packaging experience.

### CODE REVIEW

**Comment by doron-cohen** [comment]
Hey @CyberTailor . Thanks for contributing. Will this play well with GoReleaser? Also, I added some comments, please take a look.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -27,3 +27,16 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           AUR_SSH_KEY: ${{ secrets.AUR_SSH_KEY }}
           HOMEBREW_TAP_GITHUB_TOKEN: ${{ secrets.HOMEBREW_TAP_GITHUB_TOKEN }}
+
+      - name: Generate release tarball
+        run: |
+          VERSION=${GITHUB_REF_NAME#v}
```
Comment (doron-cohen) [inline .github/workflows/release.yml:33]: Can you explain the syntax?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -27,3 +27,16 @@ jobs:
...
+          tar czf antidot-${VERSION}.tar.gz --transform "s,^,antidot-${VERSION}/," --exclude dist *
+
+      - name: Upload release tarball
+        uses: softprops/action-gh-release@master
```
Comment (doron-cohen) [inline .github/workflows/release.yml:37]: I prefer to pin actions to a certain version. I saw [v0.1.14](https://github.com/softprops/action-gh-release/tree/v0.1.14) is the latest version of this action. Could you change to it?

**Review by doron-cohen — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (CyberTailor) [reply .github/workflows/release.yml:33]: `${GITHUB_REF_NAME#v}` removes "v" prefix from version for convenience

File: .github/workflows/release.yml (hunk 2 above)
Comment (CyberTailor) [reply .github/workflows/release.yml]: fixed

**Pull request merged**